src/functional_tests.rs

```diff
@@ -0,0 +1,117 @@
+// Copyright 2019 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use crate::{
+    actions::*,
+    state::*,
+    utilities::{
+        Candidate, Node, NodeState, ParsecVote, RelocatedInfo, Rpc, State, TestEvent, TryResult,
+    },
+};
+use rand::{self, Rng, SeedableRng};
+use rand_xorshift::XorShiftRng;
+use std::{env, iter, thread};
+use unwrap::unwrap;
+
+fn get_rng() -> XorShiftRng {
+    let env_var_name = "ROUTING_MODEL_SEED";
+    let seed = env::var(env_var_name)
+        .ok()
+        .map(|value| {
+            unwrap!(
+                value.parse::<u64>(),
+                "Env var 'ROUTING_MODEL_SEED={}' is not a valid u64.",
+                value
+            )
+        })
+        .unwrap_or_else(rand::random);
+    println!(
+        "To replay this '{}', set env var {}={}",
+        unwrap!(thread::current().name()),
+        env_var_name,
+        seed
+    );
+    XorShiftRng::seed_from_u64(seed)
+}
+
+#[test]
+fn relocate_adult_src() {
+    let adult_count = 2;
+    let elder_count = 3;
+    let mut rng = get_rng();
+    let mut nodes = iter::repeat_with(|| rng.gen())
+        .take(adult_count + elder_count)
+        .collect::<Vec<Node>>();
+    nodes.sort_unstable();
```

Comment:
.sorted() ? (why unstable?)

Reply:
Well, I won't need this any more after your change suggested above, but I normally default to `sort_unstable()` unless I need a stable sort since it's more efficient.

Reply:
I tend to have the opposite reasoning, if not proven to be needed for efficiency I use simple 'sort' (in C++ as well), shorter (probably intended to be default), less surprising for people, not question of is it ok to be unstable...

Reply:
My preference isn't hugely strong, but I'm not sure I agree that `sort()` is intended to be the default.  From [the docs](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.sort):

> When applicable, unstable sorting is preferred because it is generally faster than stable sorting and it doesn't allocate auxiliary memory.